docs/source/app-dev/app-arch.rst

```diff
@@ -148,6 +148,29 @@ To use command deduplication, you should:
 
 For more details on command deduplication, see the :ref:`Ledger API Services <command-submission-service-deduplication>` documentation.
 
+
+.. _handling-participant-node-failover:
+
+Handling participant node failover
+**********************************
+
+Some DAML Ledgers support exposing multiple eventually consistent Ledger API endpoints
+where command deduplication works across the Ledger API endpoints,
+e.g., by hosting separate participant nodes that expose the same view onto the ledger.
+Contact your ledger operator to find out whether this applies to your ledger.
+
+On such ledgers, applications can easily switch between Ledger API endpoints to handle participant node failovers.
+To support switching to a different ledger API endpoint at run time,
+your application should keep track of the last ledger offset received from the :ref:`transaction service <transaction-service>`.
+After switching to a new Ledger API endpoint, subscribe to the transaction stream using the last received offset.
+
+The subscription may return a OUT_OF_RANGE error.
```

Comment:
- add example for message
- add explanation that this is due to the eventual consistency. The new Ledger API endpoint might be behind the offset requested, but it will catch up eventually.